lntest/itest/lnd_neutrino_test.go

```diff
@@ -0,0 +1,45 @@
+package itest
+
+import (
+	"context"
+
+	"github.com/lightningnetwork/lnd/lnrpc/neutrinorpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/require"
+)
+
+// testNeutrino checks that the neutrino sub-server can fetch compact
+// block filters, server status and connect to a connected peer.
+func testNeutrino(net *lntest.NetworkHarness, t *harnessTest) {
+	if net.BackendCfg.Name() != lntest.NeutrinoBackendName {
+		t.Skipf("skipping test for non neutrino backends")
+	}
+
+	ctxb := context.Background()
+
+	// Check if the neutrino sub server is running.
+	statusReq := &neutrinorpc.StatusRequest{}
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	statusRes, err := net.Alice.NeutrinoClient.Status(ctxt, statusReq)
+	require.NoError(t.t, err, "unable to receive the status from the "+
+		"neutrino sub-server")
+	require.True(t.t, statusRes.Active)
+	require.NotEqual(t.t, len(statusRes.Peers), 0,
+		"unable to find any peers")
+
+	// Request the compact block filter of the best block.
+	cFilterReq := &neutrinorpc.GetCFilterRequest{
+		Hash: statusRes.GetBlockHash(),
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	_, err = net.Alice.NeutrinoClient.GetCFilter(ctxt, cFilterReq)
+	require.NoError(t.t, err, "unable to request a compact block filter")
+
+	// Try to reconnect to a connected peer.
+	addPeerReq := &neutrinorpc.AddPeerRequest{
+		PeerAddrs: statusRes.Peers[0],
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	_, err = net.Alice.NeutrinoClient.AddPeer(ctxt, addPeerReq)
```

Comment:
We (or to be more precise, @yyforyongyu) created a new itest framework in `lntemp` that started to extract all RPC calls to individual files (for example `lntemp/rpc/wallet_kit.go`) that then handle the timeout and error path. Since this PR has been around for quite a while I wouldn't say it's a blocker to port it over to the new framework, but definitely a "nice to have".

Reply:
yeah def nice to have. Didn't mention it as it'd be quicker for me to make the change, so feel free @ErikEk if you want to invest some time to get familiar with the upcoming new test framework!

Reply:
@yyforyongyu Working on it right now!